src/Entity/StaticServiceProviderRepository.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Surfnet\SamlBundle\Entity;
+
+use Surfnet\SamlBundle\Entity\ImmutableCollection\ServiceProviders;
+use Surfnet\SamlBundle\Exception\NotFound;
+
+final class StaticServiceProviderRepository implements ServiceProviderRepository
+{
+    private $serviceProviders;
+
+    /**
+     *
+     * @param ServiceProvider[] $serviceProviders
+     */
+    public function __construct(array $serviceProviders)
+    {
+        $this->serviceProviders = new ServiceProviders($serviceProviders);
+    }
+
+    /**
+     * @param string $entityId
+     * @return ServiceProvider
+     * @throws \Surfnet\SamlBundle\Exception\NotFound
```

Comment:
Out of curiosity: what is your rationale for using a FQN for the `@throws` but not for the `@return` annotation?